rotkehlchen/chain/evm/decoding/interfaces.py

```diff
@@ -171,8 +171,64 @@ def _decode_vote_cast(self, context: DecoderContext) -> DecodingOutput:
 
 class ReloadableDecoderMixin(metaclass=ABCMeta):
 
+    def __init__(
+            self,
+            cache_type_to_check_for_freshness: CacheType,
+            query_data_method: Callable,
+            save_data_to_cache_method: Callable,
+            read_pools_and_gauges_from_cache_method: Callable,
+    ) -> None:
+        """
+        :param cache_type_to_check_for_freshness: The cache type that is checked for freshness.
+        :param query_data_method: The method that queries the remote source for data.
+        :param save_data_to_cache_method: The method that saves the data to the cache tables.
+        :param read_pools_and_gauges_from_cache_method: The method that reads the data from the
+        cache tables.
+        """
+        self.cache_type_to_check_for_freshness = cache_type_to_check_for_freshness
+        self.query_data_method = query_data_method
+        self.save_data_to_cache_method = save_data_to_cache_method
+        self.read_pools_and_gauges_from_cache_method = read_pools_and_gauges_from_cache_method
+        self.pools, self.gauges = self.read_pools_and_gauges_from_cache_method()
+
     @abstractmethod
+    def _decode_pool_events(self, context: DecoderContext) -> DecodingOutput:
+        """Subclasses should implement this method. It decodes pool related events."""
```

Comment:
This comment is a bit useless since the abstractmethod decorator already tells you that needs to be implemented.

```suggestion
        """Decodes events related to protocol pools""
```

Same for the gauges